# The main entry point of your workflow.
# After configuring, running snakemake -n in a clone of this repository should successfully execute a dry-run of the workflow.

include: "rules/common.smk"
configfile: "config.yaml"


rule all:
    input:
        # The first rule should define the default target files
        # Subsequent target rules can be specified below. They should start with all_*.
        # SRA_fastq_for_download,
        # expand("Sample_Fastq/{sample}.fastq.gz", sample=SampleList),
        # expand( "Alignments/SecondPass/{sample}/SJ.out.annotated.tab", sample=SampleList),
        config["Human_ref"]["genome_dir_hisat"] + "genome.1.ht2",
        expand( "Alignments/SecondPass/{sample}/Unstranded.bw", sample=SampleList),
        expand(config["gitinclude_output"] + "SJoutAnnotatedAndIntersected/{sample}.tab.gz", sample=SampleList),
        expand("UnstrandedBigwigs/{sample}.bw", sample=SampleList),
        expand("Junctions/{sample}.junctions.bed.gz", sample=SampleList),
        expand("Alignments/SecondPass_PE/{sample}/Aligned.sortedByCoord.out.bam", sample=SampleList_PE),
        "Misc/computeMatrix.gz",
        # "Misc/computeMatrix.introns.gz",
        "Misc/Gm12878.chromHMM.hg19.bed.gz",
        # "leafcutter/leafcutter_perind.counts.gz"
        "../../data/Gm12878.chromHMM.hg38.bed.gz"

include: "rules/other.smk"
include: "rules/DownloadAndMergeFastq.smk"
include: "rules/Align.smk"
