<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Ben Fair" />


<title>20190909_Count3ssRatioExample</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">cheRNA_pilot</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">20190909_Count3ssRatioExample</h1>
<h4 class="author">Ben Fair</h4>
<h4 class="date">9/10/2019</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-09-16
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>cheRNA_pilot/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190813code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190813)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190813code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190813)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkuncheRNApilottree4ea43f8c9ae683deffd93906eab920028e8a1bc5targetblank4ea43f8a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/cheRNA_pilot/tree/4ea43f8c9ae683deffd93906eab920028e8a1bc5" target="_blank">4ea43f8</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkuncheRNApilottree4ea43f8c9ae683deffd93906eab920028e8a1bc5targetblank4ea43f8a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Unstaged changes:
    Modified:   analysis/20190909_Count3ssRatioExample.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/cheRNA_pilot/blob/4ea43f8c9ae683deffd93906eab920028e8a1bc5/analysis/20190909_Count3ssRatioExample.Rmd" target="_blank">4ea43f8</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-09-16
</td>
<td>
update 3ss quantification analyses
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/cheRNA_pilot/4ea43f8c9ae683deffd93906eab920028e8a1bc5/docs/20190909_Count3ssRatioExample.html" target="_blank">4ea43f8</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-09-16
</td>
<td>
update 3ss quantification analyses
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/cheRNA_pilot/blob/78712fad5fba33488c8b90887697ef9698bb9a6e/analysis/20190909_Count3ssRatioExample.Rmd" target="_blank">78712fa</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-09-16
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/cheRNA_pilot/78712fad5fba33488c8b90887697ef9698bb9a6e/docs/20190909_Count3ssRatioExample.html" target="_blank">78712fa</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-09-16
</td>
<td>
update site
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)</code></pre>
<p>For each 3’ss, I used bedtools to count the number of reads overlapping the upstream 25bp window, and the downstream 25bp window. The ratio of these of these two counts is a measure of splicing efficiency, and the ratio of those ratios in nascent RNA-seq versus standard RNA-seq is a measure of how contranscriptional a splicing event is (see <a href="https://www.ncbi.nlm.nih.gov/pubmed/25929182">Herzel et al</a>). Here I will look at these metrics using a standard RNA-seq dataset and nascent RNA-seq dataset from the same LCL line 18862.</p>
<p>I have already calculated counts in upstream and downstream windows for each annotated 3’ss using bedtools (see <a href="https://github.com/bfairkun/cheRNA_pilot/tree/master/code/snakemake_workflow">Snakemake code</a>). Here I will investigate the results with a few plots. One result I expect to see is that alernative 3’ss will be spliced less contranscripionally. In the case of cassette exons, it is important to distinguish between the upstream 3’ss and the downstream 3’ss. Upstream 3’ss introns will naturally have less coverage in the downstream window (i.e. in the cassette exon).</p>
<p>First read in Gencode annotated introns and classify the 3’ss as alternative or constitutive, based on how many introns are annotated with each 3’ss. This Gencode file of introns was made using <code>gtf2leafcutter.pl</code> script in leafcutter github repo, and it is a redundant file of all intron segments from Gencode gtf. In other words, there are many overlapping introns, and many with the same 3’ss (or 5’ss). So here is the pseudo-code logic on how I classify introns:</p>
<p>Constitutive introns: &gt; 3’ss is paired to one and only one 5’ss in the annotations</p>
<p>Downstream alternative introns: &gt; 3’ss is paired to multiple 5’ss in the annotations, and all of the 5’ss that it is paired to are only paired to that 3’ss. This will also include 3’ss that are associated with alt 5’ss usage.</p>
<p>Upstream alternative introns: &gt; 3’ss is paired to a 5’ss that is used in more than 1 intron. This will also include 3’ss that are associated with alt 3’ss usage.</p>
<pre class="r"><code>GencodeIntrons &lt;- read.table(&quot;../data/GencodeHg38_all_introns.corrected.bed.gz&quot;, sep=&#39;\t&#39;, col.names = c(&#39;chrom&#39;, &#39;start&#39;, &#39;stop&#39;, &#39;name&#39;, &#39;score&#39;, &#39;strand&#39;, &#39;gene&#39;, &#39;intronNumber&#39;, &#39;transcriptType&#39;))

SpliceSitesCounted &lt;- GencodeIntrons %&gt;%
  filter(transcriptType==&quot;protein_coding&quot;) %&gt;%
  distinct(chrom, start, stop, .keep_all = T) %&gt;%
  mutate(Acceptor = case_when(strand == &quot;+&quot; ~ paste(chrom, stop, strand, sep=&quot;.&quot;),
                              strand == &quot;-&quot; ~ paste(chrom, start, strand, sep=&quot;.&quot;))) %&gt;%
  mutate(Donor = case_when(strand == &quot;+&quot; ~ paste(chrom, start, strand, sep=&quot;.&quot;),
                           strand == &quot;-&quot; ~ paste(chrom, stop, strand, sep=&quot;.&quot;))) %&gt;%
  add_count(Acceptor, name=&quot;AcceptorCount&quot;) %&gt;%
  add_count(Donor, name=&quot;DonorCount&quot;)

Annotated &lt;- SpliceSitesCounted %&gt;%
  # filter(AcceptorCount==1 &amp; DonorCount==1) %&gt;% dim()
  group_by(Acceptor) %&gt;%
  mutate(DonorMax=max(DonorCount)) %&gt;%
  ungroup() %&gt;%
  distinct(Acceptor, .keep_all = T) %&gt;%
  mutate(Type3ss = case_when(AcceptorCount == 1 &amp; DonorMax==1 ~ &quot;Constitutive intron&quot;,
                             AcceptorCount &gt;1 &amp; DonorMax==1 ~ &quot;Downstream alternative intron&quot;,
                             DonorMax &gt; 1 ~ &quot;Upstream alternative intron&quot;))

table(Annotated$Type3ss)</code></pre>
<pre><code>
          Constitutive intron Downstream alternative intron 
                       149839                          7003 
  Upstream alternative intron 
                        51850 </code></pre>
<pre class="r"><code># Write out as bed file to visually inspect classifications in IGV to check for bugs
# Manual inspection of the resulting bed file indiciates I am classifying introns as intended.
Annotated %&gt;%
  mutate(NewName=paste(Acceptor, Type3ss, sep=&quot;.&quot;)) %&gt;%
  # mutate(NewStart = case_when(strand == &quot;+&quot; ~ stop-25,
  #                             strand == &quot;-&quot; ~ start)) %&gt;%
  # mutate(NewStop = case_when(strand == &quot;+&quot; ~ stop,
  #                          strand == &quot;-&quot; ~ start-25)) %&gt;%
  # select(chrom, NewStart, NewStop, paste(Acceptor, Type3ss), &quot;.&quot;, strand) %&gt;% head()
  select(chrom, start, stop, NewName, score, strand) %&gt;%
  write.table(&quot;~/Temporary/3ssAnnotations.bed&quot;, quote=F, sep=&#39;\t&#39;, col.names=F, row.names = F)</code></pre>
<p>Now, similarly to Herzel et al, set a minimum read count threshold cutoff for reliable 3’ss splice ratios to use for downstream analysis. Then do some plotting. More precisely, I will plot Intron retenetion percent spliced in (Intron retention PSI) which will be <span class="math inline">\(IntronRetentionPSI=\frac{UpstreamCoverage}{DownstreamCoverage + UpstreamCoverage}\)</span>.</p>
<pre class="r"><code>Cutoff &lt;- 50

RS &lt;- read.table(&quot;../output/3ssCoverageBeds/NA18862_argonne.bed.gz&quot;, sep=&#39;\t&#39;, col.names = c(&#39;chrom&#39;, &#39;start&#39;, &#39;stop&#39;, &#39;name&#39;, &#39;score&#39;, &#39;strand&#39;, &#39;upstream&#39;, &#39;downstream&#39;), stringsAsFactors = F) %&gt;% filter(upstream + downstream &gt;= Cutoff)
nRS &lt;- read.table(&quot;../output/3ssCoverageBeds/18862_cheRNA_1.bed.gz&quot;, sep=&#39;\t&#39;, col.names = c(&#39;chrom&#39;, &#39;start&#39;, &#39;stop&#39;, &#39;name&#39;, &#39;score&#39;, &#39;strand&#39;, &#39;upstream&#39;, &#39;downstream&#39;), stringsAsFactors = F) %&gt;% filter(upstream + downstream &gt;= Cutoff)

ToPlot &lt;- RS %&gt;%
  left_join(nRS, by=&quot;name&quot;) %&gt;%
  inner_join(Annotated, by=c(&quot;name&quot; = &quot;Acceptor&quot;))%&gt;%
  mutate(RS.ratio = upstream.x/(downstream.x+upstream.x)) %&gt;%
  mutate(nRS.ratio = upstream.y/(downstream.y+upstream.y)) %&gt;%
  mutate(NormalizedRatio = nRS.ratio/RS.ratio)

# Plot Intron retenetion PSI.
ggplot(ToPlot, aes(color=Type3ss)) +
  stat_ecdf(aes(x=nRS.ratio, linetype=&quot;nascent-RNA&quot;), geom = &quot;step&quot;) +
  stat_ecdf(aes(x=RS.ratio, linetype=&quot;polyA-RNA&quot;), geom = &quot;step&quot;) +
  ylab(&#39;Cumulative fraction&#39;) +
  xlab(&quot;Intron retention PSI&quot;) +
  theme_bw()</code></pre>
<p><img src="figure/20190909_Count3ssRatioExample.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/cheRNA_pilot/blob/78712fad5fba33488c8b90887697ef9698bb9a6e/docs/figure/20190909_Count3ssRatioExample.Rmd/unnamed-chunk-2-1.png" target="_blank">78712fa</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-09-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Chromatin associated RNA has higher <span class="math inline">\(Intron Retention PSI\)</span> as expected. Alternative upstream introns also have <span class="math inline">\(Intron Retention PSI\)</span> as expected, since there should be less coverage downstream of those 3’ss for cassette exons. Now quantify cotrnanscriptional splicing with a metric defined as <span class="math inline">\(\frac{Intron retention PSI_{nascent}}{Intron retention PSI_{polyA}}\)</span>.</p>
<pre class="r"><code># Plot cotranscriptional splicing
ggplot(ToPlot, aes(x=NormalizedRatio, color=Type3ss)) +
  stat_ecdf(geom = &quot;step&quot;) +
  ylab(&#39;Cumulative fraction&#39;) +
  xlab(&quot;Cotranscriptional splicing ratio&quot;) +
  scale_x_continuous(trans=&#39;log10&#39;) +
  theme_bw()</code></pre>
<p><img src="figure/20190909_Count3ssRatioExample.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/cheRNA_pilot/blob/4ea43f8c9ae683deffd93906eab920028e8a1bc5/docs/figure/20190909_Count3ssRatioExample.Rmd/unnamed-chunk-3-1.png" target="_blank">4ea43f8</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-09-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ToTest &lt;- ToPlot %&gt;%
  filter(Type3ss %in% c(&quot;Constitutive intron&quot;, &quot;Downstream alternative intron&quot;))
wilcox.test(NormalizedRatio ~ Type3ss, data=ToTest)</code></pre>
<pre><code>
    Wilcoxon rank sum test with continuity correction

data:  NormalizedRatio by Type3ss
W = 134980, p-value = 0.0002172
alternative hypothesis: true location shift is not equal to 0</code></pre>
<p>As expected from <a href="https://www.ncbi.nlm.nih.gov/pubmed/25929182">Herzel et al</a>, 3’ss downstream of cassette exons (which are an intron-centric measure of splicing of the upstream cassette exon) are spliced slower than constitutive introns.</p>
<p>Next I should see how replicates compare, to see if this statistic is reasonably stable (eventually I could consider QTL mapping using this statistic).</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.14

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] forcats_0.4.0   stringr_1.4.0   dplyr_0.8.1     purrr_0.3.2    
[5] readr_1.3.1     tidyr_0.8.3     tibble_2.1.3    ggplot2_3.1.1  
[9] tidyverse_1.2.1

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.1       cellranger_1.1.0 plyr_1.8.4       pillar_1.4.1    
 [5] compiler_3.5.1   git2r_0.25.2     workflowr_1.4.0  tools_3.5.1     
 [9] digest_0.6.19    lubridate_1.7.4  jsonlite_1.6     evaluate_0.14   
[13] nlme_3.1-140     gtable_0.3.0     lattice_0.20-38  pkgconfig_2.0.2 
[17] rlang_0.3.4      cli_1.1.0        rstudioapi_0.10  yaml_2.2.0      
[21] haven_2.1.0      xfun_0.7         withr_2.1.2      xml2_1.2.0      
[25] httr_1.4.0       knitr_1.23       hms_0.4.2        generics_0.0.2  
[29] fs_1.3.1         rprojroot_1.3-2  grid_3.5.1       tidyselect_0.2.5
[33] glue_1.3.1       R6_2.4.0         readxl_1.3.1     rmarkdown_1.13  
[37] modelr_0.1.4     magrittr_1.5     whisker_0.3-2    backports_1.1.4 
[41] scales_1.0.0     htmltools_0.3.6  rvest_0.3.4      assertthat_0.2.1
[45] colorspace_1.4-1 labeling_0.3     stringi_1.4.3    lazyeval_0.2.2  
[49] munsell_0.5.0    broom_0.5.2      crayon_1.3.4    </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
