<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Ben Fair" />


<title>20190805_PlotIntronPositions</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">cheRNA_pilot</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">20190805_PlotIntronPositions</h1>
<h4 class="author">Ben Fair</h4>
<h4 class="date">8/15/2019</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-09-16
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>cheRNA_pilot/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190813code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190813)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190813code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190813)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkuncheRNApilottree68df993b35b543c75aa946291fc809f0dde8ec9ftargetblank68df993a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/cheRNA_pilot/tree/68df993b35b543c75aa946291fc809f0dde8ec9f" target="_blank">68df993</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkuncheRNApilottree68df993b35b543c75aa946291fc809f0dde8ec9ftargetblank68df993a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  analysis/20190909_Count3ssRatioExample.Rmd

Unstaged changes:
    Modified:   analysis/20190805_PlotIntronPositions.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/cheRNA_pilot/blob/9b62d8b5d27e2c9cdce639507a09312e056752d4/analysis/20190805_PlotIntronPositions.Rmd" target="_blank">9b62d8b</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-08-26
</td>
<td>
addded to site, added scripts
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/cheRNA_pilot/9b62d8b5d27e2c9cdce639507a09312e056752d4/docs/20190805_PlotIntronPositions.html" target="_blank">9b62d8b</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-08-26
</td>
<td>
addded to site, added scripts
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre class="r"><code>GencodeIntrons &lt;- read.table(&quot;../data/GencodeHg38_all_introns.corrected.bed.gz&quot;, sep=&#39;\t&#39;, col.names = c(&#39;chrom&#39;, &#39;start&#39;, &#39;stop&#39;, &#39;name&#39;, &#39;score&#39;, &#39;strand&#39;, &#39;gene&#39;, &#39;intronNumber&#39;, &#39;transcriptType&#39;))
table(GencodeIntrons$transcriptType)</code></pre>
<pre><code>
          3prime_overlapping_ncRNA                          antisense 
                                44                              24223 
     bidirectional_promoter_lncRNA                          IG_C_gene 
                               938                                 76 
                   IG_C_pseudogene                          IG_V_gene 
                                 6                                143 
                   IG_V_pseudogene                            lincRNA 
                               100                              32092 
                        non_coding                     non_stop_decay 
                                 4                                538 
           nonsense_mediated_decay             polymorphic_pseudogene 
                            123432                                280 
              processed_pseudogene               processed_transcript 
                              1621                              99728 
                    protein_coding                         pseudogene 
                            665451                                 56 
                   retained_intron                     sense_intronic 
                             94262                                776 
                 sense_overlapping                                TEC 
                               719                                  9 
                         TR_C_gene                          TR_V_gene 
                                17                                105 
                   TR_V_pseudogene   transcribed_processed_pseudogene 
                                24                                167 
    transcribed_unitary_pseudogene transcribed_unprocessed_pseudogene 
                               922                               4700 
                unitary_pseudogene             unprocessed_pseudogene 
                               197                               5449 </code></pre>
<pre class="r"><code>NMD.Transcript.Introns &lt;- GencodeIntrons %&gt;%
  mutate(Intron=paste(chrom, start,stop,strand, sep=&quot;.&quot;)) %&gt;%
  filter(transcriptType==&quot;nonsense_mediated_decay&quot;) %&gt;%
  distinct(Intron) %&gt;% pull(Intron)
Non.NMD.Transcript.Introns &lt;- GencodeIntrons %&gt;%
  mutate(Intron=paste(chrom, start,stop,strand, sep=&quot;.&quot;)) %&gt;%
  filter(transcriptType!=&quot;nonsense_mediated_decay&quot;) %&gt;%
  distinct(Intron) %&gt;% pull(Intron)
NMD.Specific.Introns &lt;- setdiff(NMD.Transcript.Introns, Non.NMD.Transcript.Introns)



files &lt;- list.files(path=&quot;../output/SJoutAnnotatedAndIntersected&quot;, pattern=&quot;*.tab.gz&quot;, full.names=TRUE, recursive=FALSE)
SampleName&lt;-gsub(&quot;../output/SJoutAnnotatedAndIntersected/(.+).tab.gz&quot;, &quot;\\1&quot;, files, perl=T)
SampleName</code></pre>
<pre><code> [1] &quot;18862_cheRNA_1&quot;                  &quot;18862_cheRNA_2&quot;                 
 [3] &quot;18913_cheRNA_1&quot;                  &quot;18913_cheRNA_2&quot;                 
 [5] &quot;19138_cheRNA_1&quot;                  &quot;19138_cheRNA_2&quot;                 
 [7] &quot;19160_cheRNA_1&quot;                  &quot;19160_cheRNA_2&quot;                 
 [9] &quot;19201_cheRNA_1&quot;                  &quot;19201_cheRNA_2&quot;                 
[11] &quot;CPE_1&quot;                           &quot;NA18862_argonne&quot;                
[13] &quot;NA18913_30min&quot;                   &quot;NA18913_60min&quot;                  
[15] &quot;NA18913_argonne&quot;                 &quot;NA19138_30min&quot;                  
[17] &quot;NA19138_60min&quot;                   &quot;NA19138_argonne&quot;                
[19] &quot;NA19160_argonne&quot;                 &quot;NA19201_30min&quot;                  
[21] &quot;NA19201_60min&quot;                   &quot;NA19201_argonne&quot;                
[23] &quot;SNE_1&quot;                           &quot;Sultan_polyA_Total&quot;             
[25] &quot;Sultan_rRNADepelete_cytoplasmic&quot; &quot;Sultan_rRNADepelete_nuclear&quot;    
[27] &quot;Sultan_rRNADeplete_Total&quot;       </code></pre>
<pre class="r"><code>#Create some translations for nicer labels
SampleName</code></pre>
<pre><code> [1] &quot;18862_cheRNA_1&quot;                  &quot;18862_cheRNA_2&quot;                 
 [3] &quot;18913_cheRNA_1&quot;                  &quot;18913_cheRNA_2&quot;                 
 [5] &quot;19138_cheRNA_1&quot;                  &quot;19138_cheRNA_2&quot;                 
 [7] &quot;19160_cheRNA_1&quot;                  &quot;19160_cheRNA_2&quot;                 
 [9] &quot;19201_cheRNA_1&quot;                  &quot;19201_cheRNA_2&quot;                 
[11] &quot;CPE_1&quot;                           &quot;NA18862_argonne&quot;                
[13] &quot;NA18913_30min&quot;                   &quot;NA18913_60min&quot;                  
[15] &quot;NA18913_argonne&quot;                 &quot;NA19138_30min&quot;                  
[17] &quot;NA19138_60min&quot;                   &quot;NA19138_argonne&quot;                
[19] &quot;NA19160_argonne&quot;                 &quot;NA19201_30min&quot;                  
[21] &quot;NA19201_60min&quot;                   &quot;NA19201_argonne&quot;                
[23] &quot;SNE_1&quot;                           &quot;Sultan_polyA_Total&quot;             
[25] &quot;Sultan_rRNADepelete_cytoplasmic&quot; &quot;Sultan_rRNADepelete_nuclear&quot;    
[27] &quot;Sultan_rRNADeplete_Total&quot;       </code></pre>
<pre class="r"><code>MergedData &lt;- data.frame()
for (i in seq_along(files)){
    CurrentDataset &lt;- read.table(files[i], sep=&#39;\t&#39;, col.names=c(&quot;chr&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;ASType&quot;, &quot;geneChr&quot;, &quot;geneStart&quot;, &quot;geneStop&quot;, &quot;gene&quot;, &quot;geneScore&quot;, &quot;geneStrand&quot;, &quot;Overlap&quot;)) %&gt;%
      select(ASType, start, stop, geneChr, geneStart, geneStop, score, strand) %&gt;%
      mutate(samplename=SampleName[i]) %&gt;%
      filter(geneChr != &quot;.&quot;) %&gt;%
      mutate(Rel5SplicePos = case_when(
        strand==&quot;+&quot; ~ ((start-geneStart)/(geneStop-geneStart)),
        strand==&quot;-&quot; ~ ((geneStop-stop)/(geneStop-geneStart))
      )) %&gt;%
      mutate(Rel3SplicePos = case_when(
        strand==&quot;+&quot; ~ ((stop-geneStart)/(geneStop-geneStart)),
        strand==&quot;-&quot; ~ ((geneStop-start)/(geneStop-geneStart))
      )) %&gt;%
      filter(Rel5SplicePos&lt;=1 &amp; Rel5SplicePos&gt;=0) %&gt;%
      filter(Rel3SplicePos&lt;=1 &amp; Rel3SplicePos&gt;=0)
    MergedData&lt;-rbind(MergedData,CurrentDataset)
}

SamplesToPlot &lt;- c(&quot;Sultan_polyA_Total&quot;, &quot;Sultan_rRNADeplete_Total&quot;, &quot;Sultan_rRNADepelete_nuclear&quot;, &quot;Sultan_rRNADepelete_cytoplasmic&quot;, &quot;18862_cheRNA_1&quot;,&quot;NA18862_argonne&quot;, &quot;19201_cheRNA_1&quot;, &quot;18913_cheRNA_2&quot;)

# SamplesToPlot &lt;- c(&quot;CPE_1&quot;, &quot;SNE_1&quot;, &quot;18862_cheRNA_1&quot;,&quot;NA18862_argonne&quot;, &quot;19201_cheRNA_1&quot;, &quot;18913_cheRNA_2&quot;)

# SamplesToPlot &lt;- c(&quot;Sultan_polyA_Total&quot;, &quot;Sultan_rRNADeplete_Total&quot;, &quot;Sultan_rRNADepelete_nuclear&quot;, &quot;Sultan_rRNADepelete_cytoplasmic&quot;)


MergedData %&gt;%
  group_by(samplename, ASType) %&gt;%
  summarise(a_sum=sum(score)) %&gt;%
  group_by(samplename) %&gt;%
  mutate(FractionSpliceType = a_sum/sum(a_sum)*100) %&gt;%
  filter(ASType!=&quot;AnnotatedSpliceSite&quot;) %&gt;%
  ggplot(aes(x = samplename, y = FractionSpliceType, fill = ASType)) +
  geom_col() +
  ylab(&quot;Percent unannotated splicing&quot;) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust=0.5))</code></pre>
<p><img src="figure/20190805_PlotIntronPositions.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/cheRNA_pilot/blob/9b62d8b5d27e2c9cdce639507a09312e056752d4/docs/figure/20190805_PlotIntronPositions.Rmd/unnamed-chunk-2-1.png" target="_blank">9b62d8b</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#Same plot but plot percent NMD specific splicing
MergedData %&gt;%
  mutate(Intron=paste(geneChr, start,stop,strand, sep=&quot;.&quot;)) %&gt;%
  mutate(NMD.status = case_when(Intron %in% NMD.Specific.Introns ~ &quot;NMD.Specific.Intron&quot;,
                                !Intron %in% NMD.Specific.Introns ~ &quot;Not.NMD.Specific.Intron&quot;
                                )) %&gt;%
  group_by(samplename, NMD.status) %&gt;%
  summarise(a_sum=sum(score)) %&gt;%
  group_by(samplename) %&gt;%
  mutate(FractionSpliceType = a_sum/sum(a_sum)*100) %&gt;%
  filter(NMD.status==&quot;NMD.Specific.Intron&quot;) %&gt;%
  ggplot(aes(x = samplename, y = FractionSpliceType)) +
  geom_col() +
  ylab(&quot;Percent NMD.specific\njunction reads&quot;) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust=0.5))</code></pre>
<p><img src="figure/20190805_PlotIntronPositions.Rmd/unnamed-chunk-2-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-2">
Past versions of unnamed-chunk-2-2.png
</button>
</p>
<div id="fig-unnamed-chunk-2-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/cheRNA_pilot/blob/9b62d8b5d27e2c9cdce639507a09312e056752d4/docs/figure/20190805_PlotIntronPositions.Rmd/unnamed-chunk-2-2.png" target="_blank">9b62d8b</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#Same plot but count fraction of spliced reads as fraction of total mapped
TotalReadCounts &lt;- read.table(&quot;../output/CountsPerBam.txt&quot;, header=T, sep=&#39;\t&#39;) %&gt;%
  mutate(samplename = gsub(&quot;.+SecondPass/(.+?)/Aligned.sortedByCoord.+&quot;, &quot;\\1&quot;, Filename, perl=T)) %&gt;%
  select(samplename, ReadCount)

MergedData %&gt;%
  group_by(samplename, ASType) %&gt;%
  summarise(a_sum=sum(score)) %&gt;%
  group_by(samplename) %&gt;%
  left_join(TotalReadCounts) %&gt;%
  mutate(FractionSpliceType = a_sum/ReadCount*100) %&gt;%
  ggplot(aes(x = samplename, y = FractionSpliceType)) +
  geom_col() +
  ylab(&quot;Percent spliced reads&quot;) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust=0.5))</code></pre>
<p><img src="figure/20190805_PlotIntronPositions.Rmd/unnamed-chunk-2-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-3">
Past versions of unnamed-chunk-2-3.png
</button>
</p>
<div id="fig-unnamed-chunk-2-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/cheRNA_pilot/blob/9b62d8b5d27e2c9cdce639507a09312e056752d4/docs/figure/20190805_PlotIntronPositions.Rmd/unnamed-chunk-2-3.png" target="_blank">9b62d8b</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#Where are called splice sites
MergedData %&gt;%
  filter(samplename %in% SamplesToPlot) %&gt;%
  filter(ASType!=&quot;AnnotatedSpliceSite&quot;) %&gt;%
  filter(score&gt;0) %&gt;%
ggplot(aes(x=Rel3SplicePos, color=samplename)) +
  geom_density(adjust=2, size=1) +
  xlab(&quot;Relative position of unannotated splice site&quot;) +
  theme_bw()</code></pre>
<p><img src="figure/20190805_PlotIntronPositions.Rmd/unnamed-chunk-2-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#Same, but weighted by RPM for junction, filtering out extreme outliers (&gt;10 counts)
MergedData %&gt;%
  filter(samplename %in% SamplesToPlot) %&gt;%
  filter(ASType!=&quot;AnnotatedSpliceSite&quot;) %&gt;%
  filter(score&gt;0) %&gt;%
  filter(score&lt;10) %&gt;%
  group_by(samplename) %&gt;%
  mutate(a_sum=sum(score)) %&gt;%
  ungroup() %&gt;%
ggplot(aes(x=Rel5SplicePos, color=samplename)) +
  geom_density(adjust=2, aes(weight=score/a_sum)) +
  theme_bw()</code></pre>
<p><img src="figure/20190805_PlotIntronPositions.Rmd/unnamed-chunk-2-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># MergedData %&gt;%
#   filter(samplename %in% SamplesToPlot) %&gt;%
#   filter(ASType!=&quot;AnnotatedSpliceSite&quot;) %&gt;%
#   filter(score&gt;0) %&gt;%
#   group_by(samplename) %&gt;%
#   mutate(a_sum=sum(score)) %&gt;%
#   mutate(frac=score/a_sum)%&gt;%
#   summarise(max=max(frac))</code></pre>
<p>Ok, so nuclear (and chromatin associated) fractions have more unannotated splicing by a factor of about 1.5X to 2X, and of those unannotated splice sites seem very slightly biased towards the 5’ end of genes.</p>
<p>As a control, I should make the same metaplots for annotated splice sites</p>
<pre class="r"><code>#Where are splice events
MergedData %&gt;%
  filter(samplename %in% SamplesToPlot) %&gt;%
  filter(ASType==&quot;AnnotatedSpliceSite&quot;) %&gt;%
  filter(score&gt;0) %&gt;%
  ggplot(aes(x=Rel3SplicePos, color=samplename)) +
    geom_density(adjust=2, size=1) +
    xlab(&quot;Relative position of annotated splice site&quot;) +
    theme_bw()</code></pre>
<p><img src="figure/20190805_PlotIntronPositions.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/cheRNA_pilot/blob/9b62d8b5d27e2c9cdce639507a09312e056752d4/docs/figure/20190805_PlotIntronPositions.Rmd/unnamed-chunk-3-1.png" target="_blank">9b62d8b</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#Same but weighted by junction RPM
ToPlot &lt;- MergedData %&gt;%
  filter(samplename %in% SamplesToPlot) %&gt;%
  filter(ASType==&quot;AnnotatedSpliceSite&quot;) %&gt;%
  filter(score&gt;0) %&gt;%
  filter(score&lt;500) %&gt;%
  group_by(samplename) %&gt;%
  mutate(a_sum=sum(score)) %&gt;%
  ungroup()
ggplot(ToPlot, aes(x=Rel3SplicePos, color=samplename)) +
  geom_density(adjust=2, size=1, aes(weight=score/a_sum)) +
  theme_bw()</code></pre>
<p><img src="figure/20190805_PlotIntronPositions.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-2">
Past versions of unnamed-chunk-3-2.png
</button>
</p>
<div id="fig-unnamed-chunk-3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/cheRNA_pilot/blob/9b62d8b5d27e2c9cdce639507a09312e056752d4/docs/figure/20190805_PlotIntronPositions.Rmd/unnamed-chunk-3-2.png" target="_blank">9b62d8b</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>MergedData %&gt;%
  filter(samplename %in% SamplesToPlot) %&gt;%
  filter(ASType==&quot;AnnotatedSpliceSite&quot;) %&gt;%
  filter(score&gt;0) %&gt;%
ggplot(aes(x=log10(score), color=samplename)) +
  geom_density() +
  theme_bw()</code></pre>
<p><img src="figure/20190805_PlotIntronPositions.Rmd/unnamed-chunk-3-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok, good. So the 5’ enrichment for splice sites in chromatin-associated RNAs is true for unannotated splice sites but not for annotated splice sites.</p>
<p>Ok finally, to complement the idea that this 5’ enrichment of novel splice sites has something to do with NMD decay of alternative transcripts, make some metagene plots of positions of introns that are Genocde annotated as within NMD target transcripts.</p>
<pre class="r"><code>MergedData %&gt;%
  mutate(Intron=paste(geneChr, start,stop,strand, sep=&quot;.&quot;)) %&gt;%
  mutate(NMD.status = case_when(Intron %in% NMD.Specific.Introns ~ &quot;NMD.Specific.Intron&quot;,
                                !Intron %in% NMD.Specific.Introns ~ &quot;Not.NMD.Specific.Intron&quot;
                                )) %&gt;%
  filter(samplename %in% SamplesToPlot) %&gt;%
  filter(ASType==&quot;AnnotatedSpliceSite&quot;) %&gt;%
  filter(score&lt;500) %&gt;%
  group_by(samplename) %&gt;%
  mutate(a_sum=sum(score)) %&gt;%
  ungroup() %&gt;%
ggplot(aes(x=Rel3SplicePos, color=samplename)) +
  geom_density(aes(linetype=NMD.status), adjust=2, size=1) +
  xlab(&quot;Relative position of splice site&quot;) +
  theme_bw()</code></pre>
<p><img src="figure/20190805_PlotIntronPositions.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.14

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] forcats_0.4.0   stringr_1.4.0   dplyr_0.8.1     purrr_0.3.2    
[5] readr_1.3.1     tidyr_0.8.3     tibble_2.1.3    ggplot2_3.1.1  
[9] tidyverse_1.2.1

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.1       cellranger_1.1.0 plyr_1.8.4       pillar_1.4.1    
 [5] compiler_3.5.1   git2r_0.25.2     workflowr_1.4.0  tools_3.5.1     
 [9] digest_0.6.19    lubridate_1.7.4  jsonlite_1.6     evaluate_0.14   
[13] nlme_3.1-140     gtable_0.3.0     lattice_0.20-38  pkgconfig_2.0.2 
[17] rlang_0.3.4      cli_1.1.0        rstudioapi_0.10  yaml_2.2.0      
[21] haven_2.1.0      xfun_0.7         withr_2.1.2      xml2_1.2.0      
[25] httr_1.4.0       knitr_1.23       hms_0.4.2        generics_0.0.2  
[29] fs_1.3.1         rprojroot_1.3-2  grid_3.5.1       tidyselect_0.2.5
[33] glue_1.3.1       R6_2.4.0         readxl_1.3.1     rmarkdown_1.13  
[37] modelr_0.1.4     magrittr_1.5     whisker_0.3-2    backports_1.1.4 
[41] scales_1.0.0     htmltools_0.3.6  rvest_0.3.4      assertthat_0.2.1
[45] colorspace_1.4-1 labeling_0.3     stringi_1.4.3    lazyeval_0.2.2  
[49] munsell_0.5.0    broom_0.5.2      crayon_1.3.4    </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
